plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,165 @@ func getPITRDatabaseName(database string, timestamp int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`, but exclude latest binlog
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	downloadedIndex := make(map[int]struct{})
+	for index, serverFile := range binlogFilesOnServer {
+		// We don't download the latest binlog in SyncBinlogs()
+		if serverFile.Name == latestBinlogFileOnServer.Name {
+			continue
+		}
+		for _, localFile := range binlogFilesLocal {
+			localFileName := localFile.Name()
+			if localFileName == serverFile.Name {
+				// binlog file exists on local
+				if localFile.Size() != serverFile.Size {
+					// File size not match, delete and then download it
+					if err := os.Remove(filepath.Join(saveDir, localFileName)); err != nil {
+						return fmt.Errorf("cannot remove %s, error: %w", localFileName, err)
+					}
+				} else {
+					// file size match, record it in downloadedIndex
+					downloadedIndex[index] = struct{}{}
+				}
+			}
+		}
+	}
+
+	// download the binlog files not recorded in downloadedIndex
+	for i, serverFile := range binlogFilesOnServer {
+		if _, ok := downloadedIndex[i]; ok {
+			continue
+		}
+		if err := r.downloadBinlogFile(ctx, instance, saveDir, serverFile); err != nil {
+			return fmt.Errorf("cannot sync binlog %s, error: %w", serverFile.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// SyncLatestBinlog syncs the latest binlog between the instance and `saveDir`
+func (r *Restore) SyncLatestBinlog(ctx context.Context, instance *api.Instance, saveDir string) error {
```

Comment:
Since the default binlog file size is 1GB (https://dev.mysql.com/doc/refman/5.7/en/replication-options-binary-log.html#sysvar_max_binlog_size)
Please leave a TODO here saying that we can improve by not downloading the whole latest binlog file every time, which is always increasing because of insert/update.
We may utilize the `--start-position` parameter of `mysqlbinlog` to only download from where we left, and append to the existing binlog file.

Reply:
Can be discussed. I stand with @d-bytebase , we will download the latest binlog only when doing PITR.

Reply:
SG.